### PR TITLE
[MODULAR] Overweight Quirk

### PR DESCRIPTION
**Pull request by softcerv**
## About The Pull Request
Adds in a neutral cost overweight quirk that gives the owner a movement speed debuff along with the fat trait. Additionally, if owner reaches an overweight level of nutrition, the mood debuff they are given is no longer negative.

## How This Contributes To The Skyrat Roleplay Experience
There a good chunk of people on here that play fat characters. This serves as a lightweight way to add some flavor to those characters. 

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>
  
Quirk in quirk selection
image

Quirk gain
image

Quirk loss
image

Nutrition value when the quirk is added roundstart
image

Mood event with quirk
image

Mood event remains unchanged without quirk
image

</details>

## Changelog

:cl:
add: adds in an overweight quirk that makes the owner overweight.
/:cl:

### CODE REVIEW

**Comment by Gladi-Writes** [comment]
I have a character that could really use this. Instead of relying for nutrients for it- which doesn't make much sense- her always having the 'fat' aspect applied would make more sense. 
Consider: Removing the slowdown when combined with voracious so they can keep eating quickly.

**Comment by softcerv** [comment]
Going to draft this while I implement some extra stuff.

**Comment by desertmote** [comment]
> Consider: Removing the slowdown when combined with voracious so they can keep eating quickly. 

no, this would be OP. Voracious removes the eating cap which implies infinite food healing. The fat slowdown and mood debuff is what keeps the quirk balanced.

File: modular_skyrat/master_files/code/datums/traits/neutral.dm [hunk 1]
```diff
@@ -205,3 +205,28 @@
...
+	quirk_holder.remove_movespeed_modifier(/datum/movespeed_modifier/overweight)
+
+/datum/movespeed_modifier/overweight
+	multiplicative_slowdown = 0.5 //Around that of a dufflebag, enough to be impactful but not debilitating.
+
+/datum/mood_event/fat/New(mob/parent_mob, ...)
+	. = ..()
+	if(HAS_TRAIT_FROM(parent_mob, TRAIT_FAT, QUIRK_TRAIT))
+		mood_change = 0 // They are probably used to it, no reason to be viscerally upset about it.
+		description = "<b>I'm fat.</b>"
```
Comment (GoldenAlpharex) [inline modular_skyrat/master_files/code/datums/traits/neutral.dm:232]: I still find this moodlet incredibly funny for some reason, lmao.

**Review by GoldenAlpharex — APPROVED** [review]
Seems pretty fine to me, honestly.

**Pull request merged**